pandaskill/app/leaderboard_page.py

```diff
@@ -0,0 +1,135 @@
+import streamlit as st
+import datetime as dt
+import numpy as np
+from pandaskill.experiments.skill_rating.ranking import create_global_player_ranking
+from pandaskill.experiments.general.utils import ALL_REGIONS
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+def display_leaderboard_page(data):    
+    st.header("Leaderboard")
+
+    date_default = data["date"].max()
+
+    setting_columns = st.columns(4)
+    with setting_columns[0]:
+        ranking_type = st.selectbox("Ranking type", ["Player", "Team"], 0)
+    with setting_columns[1]:
+        date = st.date_input("Date", date_default, key="leaderboard_date")
+    with setting_columns[2]:
+        all_regions_choice = ["All"] + ALL_REGIONS
+        region = st.selectbox("Region", all_regions_choice, 0)
+    with setting_columns[3]:    
+        all_roles_choice = ["All"] + data["role"].unique().tolist()
```

Comment:
is your app fast ? (I did not run it locally)
if not you could apply all filter in one short instead of doing it sequentially